analyzer/src/main/kotlin/managers/Npm.kt

```diff
@@ -301,7 +301,10 @@ open class Npm(
 
         log.info { "Searching for 'package.json' files in '$nodeModulesDir'..." }
 
-        val nodeModulesFiles = nodeModulesDir.walk().filter {
+        // PNPM symlinks workspace modules in the `node_modules` directory, which will result in cyclic symlinks of
+        // `node_module` directories. Limit the search depth to '2' in this case.
+        val searchDepth = if (managerName == "PNPM") 2 else Int.MAX_VALUE
```

Comment:
I believe that adding case distinctions for child classes to the parent class is at the minimum a code smell. 
I believe it should at least follow the open closed principle, e.g. that the behavior is injected by the child class, so that the parent class doesn't know about it. The same applies to other places below.

Apart from that, have you considered using aggregation instead of inheritance? E.g. factor out a couple of functions from NPM to NpmSupport and then make use of them from (1) Npm (2) Yarn and (3) Pnpm. This would IMO reduce complexity quite a bit and make it easy to make changes to e.g. Npm, without the need to understand all the Pnpm and Yarn details.